Queueomatic/Server/Endpoints/ForgetPassword/ForgetPasswordEndpoint.cs

```diff
@@ -0,0 +1,58 @@
+﻿using FastEndpoints;
+using Queueomatic.DataAccess.UnitOfWork;
+using Queueomatic.Server.Services.AuthenticationService;
+using Queueomatic.Server.Services.MailService;
+using Queueomatic.Shared.DTOs;
+
+
+namespace Queueomatic.Server.Endpoints.ForgetPassword;
+
+public class ForgetPasswordEndpoint : Endpoint<ForgetPasswordRequest>
+{
+    private readonly IAuthenticationService _authenticationService;
+    private readonly IConfiguration _configuration;
+    private readonly IUnitOfWork _unitOfWork;
+    private readonly IMailService _mailService;
+    public ForgetPasswordEndpoint(IAuthenticationService authenticationService, IConfiguration configuration, IUnitOfWork unitOfWork, IMailService mailService)
+    {
+        _authenticationService = authenticationService;
+        _configuration = configuration;
+        _unitOfWork = unitOfWork;
+        _mailService = mailService;
+    }
+
+    public override void Configure()
+    {
+        Post("/forgetPassword");
+        AllowAnonymous();
+    }
+
+    public override async Task HandleAsync(ForgetPasswordRequest req, CancellationToken ct)
+    {
+        var user = await _unitOfWork.UserRepository.GetAsync(req.Email);
+        if (user is null)
+        {
+            await SendNotFoundAsync();
+            return;
+        }
+
+        var token = _authenticationService.CreateRandomToken();
+        user.PasswordResetToken = token;
+        user.ResetTokenExpires = DateTime.Now.AddDays(1);
+        await _unitOfWork.SaveAsync();
+
+        var url = $"{_configuration.GetSection("MailSettings")["AppUrl"]}/ResetPasswordDto?token={user.PasswordResetToken}";
```

Comment:
I think it's much nicer and better if the reset path would be `$"{_configuration.GetSection("MailSettings")["AppUrl"]}/resetPassword/{user.PasswordResetToken}"`